sanitizer/_text_pelias_parser.js

```diff
@@ -0,0 +1,242 @@
+const logger = require('pelias-logger').get('api');
+const Tokenizer = require('pelias-parser/tokenization/Tokenizer');
+const Solution = require('pelias-parser/solver/Solution');
+const AddressParser = require('pelias-parser/parser/AddressParser');
+const parser = new AddressParser();
+const _ = require('lodash');
+
+/**
+  this module provides fulltext parsing using the pelias/parser module.
+  see: https://github.com/pelias/parser
+
+  'pelias parser' provides the following fields:
+  'name',
+  'housenumber', 'street', 'postcode',
+  'locality', 'region', 'country',
+  'admin'
+**/
+
+// validate texts, convert types and apply defaults
+function _sanitize (raw, clean) {
+  // error & warning messages
+  var messages = { errors: [], warnings: [] };
+
+  // invalid input 'text'
+  const text = _.trim(raw.text);
+  if (!_.isString(text) || _.isEmpty(text)) {
+    messages.errors.push('invalid param \'text\': text length, must be >0');
+  }
+
+  // valid input 'text'
+  else {
+    // parse text with pelias/parser
+    clean.text = text;
+    clean.parser = 'pelias';
+    clean.parsed_text = parse(clean);
+  }
+
+  return messages;
+}
+
+function parse (clean) {
+  
+  // parse text
+  let start = new Date();
+  const t = new Tokenizer(clean.text);
+  parser.classify(t);
+  parser.solve(t);
+
+  // log summary info
+  logger.info('pelias_parser', {
+    took: (new Date()) - start,
+    solutions: t.solution.length
```

Comment:
can you add more context here like in https://github.com/pelias/api/blob/5f1f32ffa138333cd362f54583ff087a5544b445/controller/search.js#L67

For example, adding the params object allows us to correlate parser info with query params. Text length is also another extremely useful computed property.

Reply:
This was fixed in 231bdc54